### ISSUE TITLE
[Bug]: Postgres charts broken on upgrade to Postgresql 17

### ISSUE THREAD

**Issue by cacraig** [netdata/netdata#18621]
### Bug description

After upgrading to recently released Postgres 17, Postgres charts stopped working. Tried restarting netdata service as well.

### Expected behavior

Postgres charts should work.

### Steps to reproduce

1. Install or Upgrade to postgres 17.
2. Charts stop working.

<img width="1155" alt="Screenshot 2024-09-26 at 4 56 06 PM" src="https://github.com/user-attachments/assets/4e7e0c89-6289-4407-91ae-014217130aa6">

### Installation method

kickstart.sh

### System info

```shell
Linux REDACTED-SERVER 6.8.0-41-generic #41-Ubuntu SMP PREEMPT_DYNAMIC Fri Aug  2 20:41:06 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=24.04
/etc/lsb-release:DISTRIB_CODENAME=noble
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 24.04.1 LTS"
/etc/os-release:PRETTY_NAME="Ubuntu 24.04.1 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION_ID="24.04"
/etc/os-release:VERSION="24.04.1 LTS (Noble Numbat)"
/etc/os-release:VERSION_CODENAME=noble
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:UBUNTU_CODENAME=noble
/etc/os-release:LOGO=ubuntu-logo
```

### Netdata build info

```shell
Packaging:
    Netdata Version ____________________________________________ : v1.47.2
    Installation Type __________________________________________ : binpkg-deb
    Package Architecture _______________________________________ : x86_64
    Package Distro _____________________________________________ : ubuntu 22.04
    Configure Options __________________________________________ : dummy-configure-command
Default Directories:
    User Configurations ________________________________________ : /etc/netdata
    Stock Configurations _______________________________________ : /usr/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /var/cache/netdata
    Permanent Databases ________________________________________ : /var/lib/netdata
    Plugins ____________________________________________________ : /usr/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /usr/share/netdata/web
    Log Files __________________________________________________ : /var/log/netdata
    Lock Files _________________________________________________ : /var/lib/netdata/lock
    Home _______________________________________________________ : /var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : Linux
    Kernel Version _____________________________________________ : 6.8.0-41-generic
    Operating System ___________________________________________ : Ubuntu
    Operating System ID ________________________________________ : ubuntu
    Operating System ID Like ___________________________________ : debian
    Operating System Version ___________________________________ : 24.04.1 LTS (Noble Numbat)
    Operating System Version ID ________________________________ : none
    Detection __________________________________________________ : /etc/os-release
Hardware:
    CPU Cores __________________________________________________ : 96
    CPU Frequency ______________________________________________ : 2300000000
    RAM Bytes __________________________________________________ : 270085656576
    Disk Capacity ______________________________________________ : 15363023929344
    CPU Architecture ___________________________________________ : x86_64
    Virtualization Technology __________________________________ : none
    Virtualization Detection ___________________________________ : systemd-detect-virt
Container:
    Container __________________________________________________ : none
    Container Detection ________________________________________ : systemd-detect-virt
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : none
    Container Operating System ID ______________________________ : none
    Container Operating System ID Like _________________________ : none
    Container Operating System Version _________________________ : none
    Container Operating System Version ID ______________________ : none
    Container Operating System Detection _______________________ : none
Features:
    Built For __________________________________________________ : Linux
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Back-filling (of higher database tiers) ____________________ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (zstd lz4 gzip)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine (compression) _____________________________________ : YES (zstd lz4)
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : YES
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    ZSTD (fast, lossless compression algorithm) ________________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Brotli (generic-purpose lossless compression algorithm) ____ : NO
    protobuf (platform-neutral data serialization protocol) ____ : YES (system)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : NO
    libcrypto (cryptographic functions) ________________________ : YES
    libyaml (library for parsing and emitting YAML) ____________ : YES
    libmnl (library for working with netfilter) ________________ : YES
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : YES
    cgroup-network (associate interfaces to CGROUPS) ___________ : YES
    proc (monitor Linux systems) _______________________________ : YES
    tc (monitor Linux network QoS) _____________________________ : YES
    diskspace (monitor Linux mount points) _____________________ : YES
    freebsd (monitor FreeBSD systems) __________________________ : NO
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : YES
    cups (monitor printers and print jobs) _____________________ : YES
    ebpf (monitor system calls) ________________________________ : YES
    freeipmi (monitor enterprise server H/W) ___________________ : YES
    nfacct (gather netfilter accounting) _______________________ : YES
    perf (collect kernel performance events) ___________________ : YES
    slabinfo (monitor kernel object caching) ___________________ : YES
    Xen ________________________________________________________ : YES
    Xen VBD Error Tracking _____________________________________ : NO
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : YES
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : YES
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
```

**Comment by ilyam8** [comment]
Hi, @cacraig. Can you provide the 

- [error](https://learn.netdata.cloud/docs/collecting-metrics/databases/postgresql#system-with-systemd)
- and [debug output](https://learn.netdata.cloud/docs/collecting-metrics/databases/postgresql#debug-mode)?

**Comment by cacraig** [comment]
```
Sep 26 22:53:24 SERVER-REDACTED netdata[979027]: level=error msg="error on pinging the Postgres database [host=/var/run/postgresql dbname=postgres user=postgres]: failed to connect to `host=/var/run/>
Sep 26 22:53:24 SERVER-REDACTED netdata[979027]: level=error msg="check failed" plugin=go.d collector=postgres job=local
Sep 26 22:53:24 SERVER netdata[979027]: level=error msg="querying database conflicts error: ERROR: column \"checkpoints_timed\" does not exist (SQLSTATE 42703)" plugin=go.d collector=postgre>
Sep 26 22:53:24 SERVER-REDACTED netdata[979027]: level=error msg="check failed" plugin=go.d collector=postgres job=local
Sep 26 22:53:24 SERVER netdata[979027]: level=error msg="error on pinging the Postgres database [postgresql://netdata@127.0.0.1:5432/postgres]: failed to connect to `host=127.0.0.1 user=netd>
Sep 26 22:53:24 SERVER-REDACTED netdata[979027]: level=error msg="check failed" plugin=go.d collector=postgres job=local
```

```
./go.d.plugin -d -m postgres
DBG godplugin/main.go:158 plugin: name=go.d, version=v1.47.2 component=agent
DBG godplugin/main.go:160 current user: name=netdata, uid=119 component=agent
INF godplugin/main.go:164 env HTTP_PROXY '', HTTPS_PROXY '' component=agent
INF agent/agent.go:145 instance is started component=agent
INF agent/setup.go:23 loading config file component=agent
DBG agent/setup.go:31 looking for 'go.d.conf' in [/etc/netdata /usr/lib/netdata/conf.d] component=agent
INF agent/setup.go:38 found '/usr/lib/netdata/conf.d/go.d.conf component=agent
INF agent/setup.go:45 config successfully loaded component=agent
INF agent/agent.go:149 using config: enabled 'true', default_run 'true', max_procs '0' component=agent
INF agent/setup.go:50 loading modules component=agent
INF agent/setup.go:73 enabled/registered modules: 1/110 component=agent
INF agent/setup.go:79 building discovery config component=agent
DBG agent/setup.go:126 looking for 'postgres.conf' in [/etc/netdata/go.d /usr/lib/netdata/conf.d/go.d] component=agent
DBG agent/setup.go:142 found '/usr/lib/netdata/conf.d/go.d/postgres.conf component=agent
INF agent/setup.go:147 dummy/read/watch paths: 0/1/0 component=agent
INF discovery/manager.go:116 registered discoverers: [file discovery: [file reader] service discovery] component="discovery manager"
DBG agent/setup.go:165 looking for 'vnodes/' in [/etc/netdata /usr/lib/netdata/conf.d] component=agent
DBG vnodes/vnodes.go:86 '/usr/lib/netdata/conf.d/vnodes' is not a regular file, skipping it component=vnodes
INF agent/setup.go:177 found '/usr/lib/netdata/conf.d/vnodes' (0 vhosts) component=agent
INF functions/manager.go:44 instance is started component="functions manager"
INF jobmgr/manager.go:87 instance is started component="job manager"
DBG functions/ext.go:15 registering function 'config' component="functions manager"
CONFIG go.d:collector:postgres create accepted template /collectors/jobs internal 'internal' 'add schema enable disable test userconfig' 0x0000 0x0000

INF discovery/manager.go:61 instance is started component="discovery manager"
INF sd/sd.go:66 instance is started component="service discovery"
INF file/discovery.go:69 instance is started component=discovery discoverer=file
INF file/read.go:48 instance is started component=discovery discoverer=file
INF file/read.go:49 instance is stopped component=discovery discoverer=file
DBG jobmgr/manager.go:122 received configs: 0/+0/-0 ('/usr/lib/netdata/conf.d/go.d/postgres.conf') component="job manager"
INF pipeline/pipeline.go:118 instance is started component="service discovery" pipeline=docker
INF dockerd/docker.go:101 instance is started component="service discovery" discoverer=docker
ERR dockerd/docker.go:118 Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running? component="service discovery" discoverer=docker
INF dockerd/docker.go:102 instance is stopped component="service discovery" discoverer=docker
INF pipeline/accumulator.go:92 discoverer 'sd:docker' exited before ctx done component="service discovery" pipeline=docker
INF pipeline/accumulator.go:61 all discoverers exited before ctx done component="service discovery" pipeline=docker
DBG pipeline/pipeline.go:139 received 0 target groups component="service discovery" pipeline=docker
INF pipeline/pipeline.go:137 instance is stopped component="service discovery" pipeline=docker
INF pipeline/pipeline.go:118 instance is started component="service discovery" pipeline="network listeners"
INF netlisteners/netlisteners.go:102 instance is started component="service discovery" discoverer=net_listeners
DBG pipeline/pipeline.go:139 received 1 target groups component="service discovery" pipeline="network listeners"
DBG pipeline/pipeline.go:154 processing group 'discoverer=net_listeners,host=localhost,file=/usr/lib/netdata/conf.d/go.d/sd/net_listeners.conf' with 16 target(s) component="service discovery" pipeline="network listeners"
DBG pipeline/compose.go:85 created 4 config(s) for target 'tcp_80_8709193245816304817'
DBG pipeline/compose.go:85 created 1 config(s) for target 'tcp_6432_16745165205271858759'
DBG pipeline/compose.go:85 created 3 config(s) for target 'tcp_5432_17634335155874022297'
DBG pipeline/compose.go:85 created 4 config(s) for target 'tcp_8080_6112052947743769253'
DBG jobmgr/manager.go:122 received configs: 12/+12/-0 ('discoverer=net_listeners,host=localhost,file=/usr/lib/netdata/conf.d/go.d/sd/net_listeners.conf') component="job manager"
CONFIG go.d:collector:postgres:local create accepted job /collectors/jobs discovered 'discoverer=net_listeners,host=localhost,file=/usr/lib/netdata/conf.d/go.d/sd/net_listeners.conf' 'schema get enable disable update restart test userconfig' 0x0000 0x0000

DBG jobmgr/manager.go:293 creating postgres[local] job, config: map[__provider__:sd:net_listeners __source__:discoverer=net_listeners,host=localhost,file=/usr/lib/netdata/conf.d/go.d/sd/net_listeners.conf __source_type__:discovered autodetection_retry:0 dsn:host=/var/run/postgresql dbname=postgres user=postgres module:postgres name:local priority:70000 update_every:1] component="job manager"
ERR postgres/postgres.go:128 error on pinging the Postgres database [host=/var/run/postgresql dbname=postgres user=postgres]: failed to connect to `host=/var/run/postgresql user=postgres database=postgres`: server error (FATAL: Peer authentication failed for user "postgres" (SQLSTATE 28000)) collector=postgres job=local
ERR module/job.go:238 check failed collector=postgres job=local
CONFIG go.d:collector:postgres:local status failed

CONFIG go.d:collector:postgres:local create accepted job /collectors/jobs discovered 'discoverer=net_listeners,host=localhost,file=/usr/lib/netdata/conf.d/go.d/sd/net_listeners.conf' 'schema get enable disable update restart test userconfig' 0x0000 0x0000

DBG jobmgr/manager.go:293 creating postgres[local] job, config: map[__provider__:sd:net_listeners __source__:discoverer=net_listeners,host=localhost,file=/usr/lib/netdata/conf.d/go.d/sd/net_listeners.conf __source_type__:discovered autodetection_retry:0 dsn:host=/var/run/postgresql dbname=postgres user=netdata module:postgres name:local priority:70000 update_every:1] component="job manager"
DBG postgres/collect.go:38 connected to PostgreSQL v170000 collector=postgres job=local
DBG postgres/collect.go:47 connected as super user: false collector=postgres job=local
DBG postgres/collect.go:56 the instance is in recovery mode: false collector=postgres job=local
ERR postgres/postgres.go:128 querying database conflicts error: ERROR: column "checkpoints_timed" does not exist (SQLSTATE 42703) collector=postgres job=local
ERR module/job.go:238 check failed collector=postgres job=local
CONFIG go.d:collector:postgres:local status failed

CONFIG go.d:collector:postgres:local create accepted job /collectors/jobs discovered 'discoverer=net_listeners,host=localhost,file=/usr/lib/netdata/conf.d/go.d/sd/net_listeners.conf' 'schema get enable disable update restart test userconfig' 0x0000 0x0000

DBG jobmgr/manager.go:293 creating postgres[local] job, config: map[__provider__:sd:net_listeners __source__:discoverer=net_listeners,host=localhost,file=/usr/lib/netdata/conf.d/go.d/sd/net_listeners.conf __source_type__:discovered autodetection_retry:0 dsn:postgresql://netdata@127.0.0.1:5432/postgres module:postgres name:local priority:70000 update_every:1] component="job manager"
ERR postgres/postgres.go:128 error on pinging the Postgres database [postgresql://netdata@127.0.0.1:5432/postgres]: failed to connect to `host=127.0.0.1 user=netdata database=postgres`: failed SASL auth (FATAL: password authentication failed for user "netdata" (SQLSTATE 28P01)) collector=postgres job=local
ERR module/job.go:238 check failed collector=postgres job=local
CONFIG go.d:collector:postgres:local status failed
```

**Comment by ilyam8** [comment]
It says password auth failed

```
ERR postgres/postgres.go:128 error on pinging the Postgres database [postgresql://netdata@127.0.0.1:5432/postgres]: failed to connect to `host=127.0.0.1 user=netdata database=postgres`: failed SASL auth (FATAL: password authentication failed for user "netdata" (SQLSTATE 28P01)) collector=postgres job=local
```

**Comment by ilyam8** [comment]
Thanks for reporting, @cacraig. Fixed in #18629.
